ui/app/AppLayouts/Chat/views/ChatColumnView.qml

```diff
@@ -234,102 +238,139 @@ Item {
-                        return
+                    onSendMessage: {
+                        if (!d.activeChatContentModule) {
+                            console.debug("error on sending message - chat content module is not set")
+                            return
+                        }
+
+                        if (root.rootStore.sendMessage(d.activeChatContentModule.getMyChatId(),
+                                                       event,
+                                                       chatInput.getTextWithPublicKeys(),
+                                                       chatInput.isReply? chatInput.replyMessageId : "",
+                                                       chatInput.fileUrlsAndSources
+                                                       ))
+                        {
+                            Global.playSendMessageSound()
+
+                            chatInput.textInput.clear();
+                            chatInput.textInput.textFormat = TextEdit.PlainText;
+                            chatInput.textInput.textFormat = TextEdit.RichText;
+                        }
                     }
 
-                    if (root.rootStore.sendMessage(d.activeChatContentModule.getMyChatId(),
-                                                  event,
-                                                  chatInput.getTextWithPublicKeys(),
-                                                  chatInput.isReply? chatInput.replyMessageId : "",
-                                                  chatInput.fileUrlsAndSources
-                                                  ))
-                    {
-                        Global.playSendMessageSound()
-
-                        chatInput.textInput.clear();
-                        chatInput.textInput.textFormat = TextEdit.PlainText;
-                        chatInput.textInput.textFormat = TextEdit.RichText;
+                    onKeyUpPress: {
+                        d.activeMessagesStore.setEditModeOnLastMessage(root.rootStore.userProfileInst.pubKey)
                     }
                 }
 
-                onKeyUpPress: {
-                    d.activeMessagesStore.setEditModeOnLastMessage(root.rootStore.userProfileInst.pubKey)
+                ChatPermissionQualificationPanel {
+                    id: channelPostRestrictions
+                    width: (chatInput.textInput.width-Style.current.bigPadding) // by design
+                    height: chatInput.textInput.height
+                    anchors.left: parent.left
+                    anchors.leftMargin: (2*Style.current.bigPadding)
+                    visible: (viewAndPostPermissionsModel.count > 0)
+                    assetsModel: root.rootStore.assetsModel
+                    collectiblesModel: root.rootStore.collectiblesModel
+                    holdingsModel: channelPostRestrictions.viewAndPostPermissionsModel
+                    //viewOnlyHoldingsModel: TODO
```

Comment:
These 3 lines can be removed now! ;)

Reply:
ain't these needed from HoldingsSelectionModel?
```
model: HoldingsSelectionModel {
    sourceModel: holdingsListModel
    assetsModel: root.assetsModel
    collectiblesModel: root.collectiblesModel
```

Reply:
I was referring to the commented lines!

Reply:
aaaah, I left those as TODO to have the placeholders for when the models are available. let me know if you think those should removed